firestore-translate-text/__tests__/functions.test.ts

```diff
@@ -0,0 +1,384 @@
+import mockedEnv from "mocked-env";
+import * as functionsTestInit from "firebase-functions-test";
+
+const testTranslations = {
+  de: "hallo",
+  en: "hello",
+  es: "hola",
+  fr: "salut",
+};
+
+const defaultEnvironment = {
+  PROJECT_ID: "extensions-testing",
+  LOCATION: "us-central1",
+  // values from extension.yaml param defaults
+  LANGUAGES: "en,es,de,fr",
+  COLLECTION_PATH: "translations",
+  INPUT_FIELD_NAME: "input",
+  OUTPUT_FIELD_NAME: "translated",
+};
+
+const mockFirestoreUpdate = jest.fn();
+
+// await translate.translate('hello', 'de');
+const mockTranslateClassMethod = jest
+  .fn()
+  .mockImplementation((string: string, targetLanguage: string) => {
+    return Promise.resolve([testTranslations[targetLanguage]]);
+  });
+
+// new Translate(opts);
+const mockTranslateClass = jest.fn().mockImplementation(() => {
+  return { translate: mockTranslateClassMethod };
+});
+
+// import { Translate } from "@google-cloud/translate";
+function mockTranslateModuleFactory() {
+  return {
+    Translate: mockTranslateClass,
+  };
+}
+
+function mockDocumentSnapshotFactory(documentSnapshot) {
+  return jest.fn().mockImplementation(() => {
+    return {
+      exists: true,
+      get: documentSnapshot.get.bind(documentSnapshot),
+      ref: {
+        path: documentSnapshot.ref.path,
+      },
+    };
+  })();
+}
+
+jest.mock("@google-cloud/translate", mockTranslateModuleFactory);
```

Comment:
This code is all tied to the mocks required for the tests.  What do you think about having a folder like `mocks` and putting this into something like a `translate.ts` file?